cime/src/share/util/shr_orb_mod.F90

```diff
@@ -62,22 +74,28 @@ real(SHR_KIND_R8) pure FUNCTION shr_orb_cosz(jday,lat,lon,declin,dt_avg,uniform_
     logical :: use_dt_avg
 
     !----------------------------------------------------------------------------
-    
+
+    if ( constant_zenith_angle_deg >= 0 ) then
+      shr_orb_cosz = cos( constant_zenith_angle_deg * SHR_CONST_PI/180. )
+      return
+    end if
+
     if (present(uniform_angle)) then
-       shr_orb_cosz = cos(uniform_angle)
-    else ! perform the calculation of shr_orb_cosz
-       use_dt_avg = .false.
-       if (present(dt_avg)) then
-          if (dt_avg /= 0.0_shr_kind_r8) use_dt_avg = .true.
-       end if
-       ! If dt for the average cosz is specified, then call the shr_orb_avg_cosz
-       if (use_dt_avg) then
-          shr_orb_cosz =  shr_orb_avg_cosz(jday, lat, lon, declin, dt_avg)
-       else
-          shr_orb_cosz = sin(lat)*sin(declin) - &
-               cos(lat)*cos(declin) * &
-               cos((jday-floor(jday))*2.0_SHR_KIND_R8*pi + lon)
-       end if
+      shr_orb_cosz = cos(uniform_angle)
+      return
```

Comment:
Really? I think it's much clearer now. I wanted the "constant_zenith_angle_deg" option to override everything. I don't like how "uniform_angle" was implemented because it requires every call to get the zenith angle to be modified. And the modification doesn't make a lot of sense because you have to get this value from the namelist and pass it down into this routine only to have it spit the same value back at you. Modifying this routine to always spit back "constant_zenith_angle_deg" when it is set doesn't require any code to be modified in E3SM. So it seems to be the simplest way over changing the zenith angle. I also needed to preserve the NCAR method of using "uniform_angle", which I think this does nicely.